cbot_guidance/src/guidance_node.cpp

```diff
@@ -1,15 +1,25 @@
-#include "ros/ros.h"
+#include <stdio.h>
 #include "geometry_msgs/Pose.h"
-#include "cbot_ros_msgs/GuidanceInputs.h"
 #include "cbot_ros_msgs/ControllerInputs.h"
+#include "cbot_ros_msgs/GuidanceInputs.h"
 #include "cbot_guidance/guidance.hpp"
+#include <time.h> 
+#include <std_msgs/Bool.h>
+#include "cbot_guidance/utm.hpp"
 
-double desired_heading, desired_thrust, nominal_velocity;
-bool guidance_status;
+int guidanceON = 0;
 
 double vehicle_pos_x, vehicle_pos_y;
-int guidance_on;
-int guidance_mode;
+double desired_heading, desired_thrust, nominal_velocity, desired_pitch;
+double desired_pos_x1,desired_pos_y1,desired_pos_x2,desired_pos_y2;
+bool guidance_status;
+int guidance_mode=-1,flag=0;
+char Zone[20];
+
+ros::ServiceServer guidance_inputs_server;
```

Comment:
ROS actions servers are better for such tasks (sending goals etc. as they also allow feedback